Cognite.Extensions/TimeSeries/TimeSeriesUpdateResultHandlers.cs

```diff
@@ -0,0 +1,138 @@
+﻿using CogniteSdk;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Cognite.Extensions
+{
+    public partial class ResultHandlers
+    {
+        private static void ParseTimeSeriesUpdateException(ResponseException ex, CogniteError err)
+        {
+            if (ex.Missing?.Any() ?? false)
+            {
+                if (ex.Message.StartsWith("Time series not found", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    err.Type = ErrorType.ItemMissing;
+                    err.Resource = ResourceType.Id;
+                    err.Values = ex.Missing.Select(dict =>
+                    {
+                        if (dict.TryGetValue("id", out var idVal) && idVal is MultiValue.Long longIdVal)
+                        {
+                            return Identity.Create(longIdVal.Value);
+                        }
+                        else if (dict.TryGetValue("externalId", out var extIdVal)
+                            && extIdVal is MultiValue.String strIdVal
+                            && strIdVal.Value != null)
+                        {
+                            return Identity.Create(strIdVal.Value);
+                        }
+                        return null!;
```

Comment:
maybe it would be better to make `CogniteError.Values` a `IEnumerable<Identity?>?`.... maybe it's too much of a hassle (and in that case we could leave it for later), but I think of `null!` as a smell that should only appear in tests and very exceptional cases.

Reply:
In principle I agree, the reason why I use it here (and everywhere in this PR, I think), is that the nullable analysis in CDF does not properly handle `Where`. The result here can obviously never be null, since we filter them out, but the compiler doesn't understand that. I _think_ this is in the pipeline, but we would be paying a price of performance to work around it ourselves, I think.

Reply:
Making Values nullable is _definitely_ too much of a hassle, and also not really a good solution, since this is exposed to users, and we _don't_ want nulls there at all.

Reply:
I have one idea, I'll see if I can use it.